### DISCUSSION TITLE
Batch merge background for every PNG files export as JPG

### DISCUSSION THREAD

**Discussion by ozhangul** [ImageMagick/ImageMagick#6792] [Help] [answered]
### ImageMagick version

7.1.1.20-Q16

### Operating system, version and so on

Windows x64

### Description

i got folder like this https://prnt.sc/3sKIjfEZhx2r

i need add back.png as background to every other png's.

i treid this code below but it can't work
`mogrify back.png *.png -composite C:\c\gg\*.png`

i tried a lot of code but i can't do it. 

thanks for help.

**Comment by ozhangul** [comment]
`convert back.jpg 2.png -compose over -composite out.jpg`
this code is working on single. but i need batch method.

thanks for helping.

**Comment by snibgo** [comment]
I suggest you put that in a shell `for` loop. For help on `for`, type `help for` at the command line.

**Reply by ozhangul** [reply]
thanks for answer.
i got that error when i did it. it says "convert was unexpected at this time."
https://prnt.sc/aFmSbC7qcdST

**Reply by ozhangul** [reply]
https://prnt.sc/ERh1N2uyd02H
i tried this command it working but i got black background. not merge with back.jpg
`for /r %g in (*.png) do magick convert "%~dpng.png" "back.jpg"  -compose CopyOpacity -composite "%~dpng.jpg"`

**Reply by ozhangul** [reply, marked as answer]
this is working for me. thanks for helping for "FOR" command.
`for /r %g in (*.png) do magick convert "back.png" "%~dpng.png"  -compose over -composite "%~dpng.jpg"`
